peerconnection.go

```diff
@@ -732,16 +773,56 @@ func (pc *PeerConnection) SetRemoteDescription(desc SessionDescription) error {
 		return err
 	}
 
-	if haveRemoteDescription {
-		pc.startRenegotation(currentTransceivers)
-		return nil
-	}
-
 	weOffer := true
 	if desc.Type == SDPTypeOffer {
 		weOffer = false
 	}
 
+	var t *RTPTransceiver
+	localTransceivers := append([]*RTPTransceiver{}, pc.GetTransceivers()...)
+	detectedPlanB := descriptionIsPlanB(pc.RemoteDescription())
+
+	if !weOffer && !detectedPlanB {
+		for _, media := range pc.RemoteDescription().parsed.MediaDescriptions {
+			midValue := getMidValue(media)
+			if midValue == "" {
+				return fmt.Errorf("RemoteDescription contained media section without mid value")
+			}
+
+			if media.MediaName.Media == "application" {
+				continue
+			}
+
+			kind := NewRTPCodecType(media.MediaName.Media)
+			direction := getPeerDirection(media)
+			if kind == 0 || direction == RTPTransceiverDirection(Unknown) {
+				continue
+			}
+
+			t, localTransceivers = findByMid(midValue, localTransceivers)
+			if t == nil {
+				t, localTransceivers = satisfyTypeAndDirection(kind, direction, localTransceivers)
+			}
+			if t == nil {
+				receiver, err := pc.api.NewRTPReceiver(kind, pc.dtlsTransport)
+				if err != nil {
+					return err
+				}
+				t = pc.newRTPTransceiver(receiver, nil, RTPTransceiverDirectionRecvonly, kind)
```

Comment:
<strike>I'm not sure if we should be adding `recvonly` transceiver here. I believe the previous code used to add an `inactive` transceiver. Adding a `recvonly` transceiver means we will blindly accept any track that the remote client gives to us. </strike>

*edit*  JSEP spec [mandates that a transceiver is created with direction `recvonly`](https://tools.ietf.org/html/draft-ietf-rtcweb-jsep-25#section-5.10):

> ```
> 5.10.  Applying a Remote Description
> 
> [...]
> 
>    For each m= section, the following steps MUST be performed; if any
>    parameters are out of bounds, or cannot be applied, processing MUST
>    stop and an error MUST be returned.
> 
>    [...]
> 
>    o  If the m= section proto value indicates use of RTP:
> 
>       *  If the m= section is not associated with any RtpTransceiver
>          (possibly because it was dissociated in the previous step),
>          either find an RtpTransceiver or create one according to the
>          following steps:
> 
>          [...]
> 
>          +  If no RtpTransceiver was found in the previous step, create
>             one with a recvonly direction.```

Does this mean we'll no longer have to call `AddTransceiver` when the remote peer wants to send a track? 

I thought the browsers require `AddTransceiver` to be called on the offer-receiving peer as well :/